apps/wallet/src/features/History/index.tsx

```diff
@@ -46,20 +46,22 @@ const HistoryFeature: FC<Props> = ({ style, network, limit }) => {
 							date = convertDateToReadable(new Date(transaction.date));
 							isOnTheSameDate = false;
 						}
-						return isOnTheSameDate ? (
-							<HistoryItem
-								key={transaction.signature}
-								transaction={transaction}
-							/>
-						) : (
-							<View
-								key={transaction.signature}
-								style={styles.dateAndTransactionContainer}
-							>
-								<Text style={styles.date}>{transactionDate}</Text>
-								<HistoryItem transaction={transaction} />
-							</View>
-						);
+
+						if ('token' in transaction)
+							return isOnTheSameDate ? (
+								<HistoryItem
+									key={transaction.signature}
+									transaction={transaction}
+								/>
+							) : (
+								<View
+									key={transaction.signature}
+									style={styles.dateAndTransactionContainer}
+								>
+									<Text style={styles.date}>{transactionDate}</Text>
+									<HistoryItem transaction={transaction} />
+								</View>
+							);
```

Comment:
not caused by this commit, but I don't think a `.map` render want to return `null`/`undefined` on its loops; as React Reconciler have no idea on how to optimize the rendering cycle.